components/date_picker/DatePicker.js

```diff
@@ -95,29 +94,79 @@ const factory = (Input, DatePickerDialog) => {
       this.setState({active: false});
     };
 
+    handleInputChange = (value) => {
+      this.setState({
+        value
+      });
+    }
+
+    formatInputText = (value, monthFirst) => {
+      const _value = monthFirst ? this.formatMonthFirstDate(value) : value;
+
+      if (this.isValidDate(_value)) {
+        const seperator = this.getSeparator(_value);
+        const dateSplit = _value.split(seperator);
+
+        return new Date(dateSplit[2], parseInt(dateSplit[1], 10) - 1, dateSplit[0]);
+      }
+
+      return value;
+    }
+
+    isValidDate = (dateString) => {
```

Comment:
This could be extracted to a `time` method helper along with other methods that are specific to text parsing to a date.